lib/rules/mixin-name-format.js

```diff
@@ -0,0 +1,78 @@
+// Note that this file is nearly identical to function-name-format.js, variable-name-format.js, and placeholder-name-format.js
+'use strict';
+
+var helpers = require('../helpers');
+
+module.exports = {
+  'name': 'mixin-name-format',
+  'defaults': {
+    'allow-leading-underscore': true,
+    'convention': 'hyphenatedlowercase'
+  },
+  'detect': function (ast, parser) {
+    var result = [];
+
+    ast.traverseByTypes(['mixin', 'include'], function (node) {
+      var name;
+
+      if (node.type === 'mixin') {
+        name = node.first('ident').content;
+      }
+      else {
+        name = node.first('simpleSelector').first('ident').content;
+      }
+
+      var originalName = name;
```

Comment:
I would also change the order here maybe..

I would probably change the order here a little for mainly flow reasons.

```
var originalName,
      name;

      if (node.type === 'mixin') {
        originalName = node.first('ident').content;
      }
      else {
        originalName = node.first('simpleSelector').first('ident').content;
      }

      name = originalName;

      if (parser.options['allow-leading-underscore'] && name[0] === '_') {
        name = name.slice(1);
      }
```

Almost identical but feels a little more consistent in flow. Ignore me if you want though :smile: